awscli/customizations/preview.py

```diff
@@ -70,15 +55,35 @@ def mark_as_preview(command_table, session, **kwargs):
     # explicitly enabled in the config file.
     allowed_services = _get_allowed_services(session)
     for preview_service in PREVIEW_SERVICES:
+        is_enabled = False
         if preview_service in allowed_services:
             # Then we don't need to swap it as a preview
             # service, the user has specifically asked to
             # enable this service.
             logger.debug("Preview service enabled through config file: %s",
                          preview_service)
-            continue
-        command_table[preview_service] = PreviewModeCommand(
-            preview_service, PREVIEW_SERVICES[preview_service])
+            is_enabled = True
+        original_command = command_table[preview_service]
+        preview_cls = type(
+            'PreviewCommand',
+            (PreviewModeCommandMixin, original_command.__class__), {})
+        command_table[preview_service] = preview_cls(
+            cli_name=original_command.name,
+            session=session,
+            service_name=original_command._service_name,
```

Comment:
I think you would would be able to get this from ``original_command.service_model.service_name``